rate_limit_quota: handle grpc_status in deny_response_settings

norrs opened the pull request:
This change ensures RLQS deny responses propagate an explicit gRPC status and
message when configured via deny_response_settings.grpc_status. For gRPC
requests, the message is used as the body text (grpc-message), aligning
behavior with other gRPC filters.

The gRPC status behavior is now simplified:
- If `grpc_status` is explicitly set, use that status code and message
- Otherwise, let Envoy derive the gRPC status from the HTTP status code

Tests: Updated/added tests in filter_test to cover grpc_status.message
Docs: Expanded RLQS filter docs to describe deny_response_settings, grpc options, with examples.
Risk Level: Low (scoped to deny path and config-driven).
Release Notes: Fixed RLQS to propagate grpc_status; clarified precedence over HTTP mapping.

norrs commented:
Docs view: 

<img width="2896" height="1940" alt="image" src="https://github.com/user-attachments/assets/320c8a66-4eec-491c-aeb1-ef182665e8eb" />

norrs commented:
For the record, I amended my commit to sign it with my correct public key associated with public Github profile. I notice I did a mistake in my other PR as mentioned here: https://github.com/envoyproxy/envoy/pull/41255#issuecomment-3348633706 , so correcting it for this one :-)

yanavlasov commented on api/envoy/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto:
```diff
@@ -251,6 +252,13 @@ message RateLimitQuotaBucketSettings {
+    // HTTP code will be 200 for a gRPC response.
+    //
+    // This field is ignored if ``grpc_status`` is explicitly set.
+    bool rate_limited_as_resource_exhausted = 5;
```
How is setting the `rate_limited_as_resource_exhausted` to `true` different from setting `grpc_status` to `RESOURCE_EXHAUSTED`

yanavlasov replied:
I see. I think the way the rate_limit filter did it is not as flexible as the rate_limit_quota. Since the behavior is identical I would just use the grpc_status since it is not that much harder.

yanavlasov reviewed (commented):
/wait-any

yanavlasov commented:
Adding @bsurber as well

yanavlasov commented:
/wait-any

norrs commented:
/wait-any

yanavlasov reviewed (commented):
/wait-any

norrs commented:
Yeah, I noticed. I have fixed it locally and removed one of the test which was no longer needed either. 

Ill nudge again after I have verified all tests etc run locally .. (my laptop is cooking thru the tests)

norrs commented:
Things are passing on CI now @yanavlasov , sorry about that. 
I still struggle a bit with running things properly on my mac, like the pre-push hooks etc.
